tests/include/detray/test/utils/simulation/event_generator/random_numbers.hpp

```diff
@@ -75,7 +75,9 @@ struct random_numbers {
 
     /// Explicit normal distribution around a @param mean and @param stddev
     DETRAY_HOST auto normal(const scalar_t mean, const scalar_t stddev) {
-        return std::normal_distribution<scalar_t>(mean, stddev)(m_engine);
+        const auto min_stddev{
+            stddev <= 0.f ? std::numeric_limits<scalar_t>::epsilon() : stddev};
+        return std::normal_distribution<scalar_t>(mean, min_stddev)(m_engine);
```

Comment:
There is nothing wrong with `stddev = 0` mathematically so we should not convert it to epsilon.
I would rather add an assertion `assert(stddev >= 0.f)` because the standard deviation must be positive.

Reply:
Fair enough, should I pass the epsilon as the default values for the vertex smearing in the track generators instead?

Reply:
Default values would be your choice. We just don't have to disallow zero stddev.

Reply:
Hm, wait, the line above also does not add up, or am I wrong? It is dividing by `q` if `q` is zero... So `q=0` should also get an assertion

Reply:
Fixed.

Reply:
What did you fix by the way

Reply:
Oh, I forgot to revert the thing in random numbers, sorry

Reply:
Done now and set the defaults to epsilon